chore: use controller-gen to generate CRDs from Go markers

erikgb opened the pull request:
## Description

Picking up https://github.com/aquasecurity/trivy-operator/pull/232, as @padlar (we are working on the same team) is not available this week, and I want to get this in as soon as possible - as it will allow me/us to suggest improvements to trivy-operator more easily. I have talked to @padlar, and got a confirmation that it's ok to finalize the changes suggested in https://github.com/aquasecurity/trivy-operator/pull/232.

This PR suggests using controller-gen to generate all CRDs from now on. After a couple of pre-PRs, I think this now looks close enough to the currently hand-coded CRDs, and possible to review.

Sadly, the labels currently added to the CRDs (version and managed-by) cannot be sustained with this change, as this is an eschewed feature in controller-gen: https://github.com/kubernetes-sigs/controller-tools/issues/454. We should be able to get them back if https://github.com/aquasecurity/trivy-operator/discussions/239 lands on a migration to kustomize as the master format.

## Related issues
- Addresses pt 2 in https://github.com/aquasecurity/trivy-operator/issues/204
- Close https://github.com/aquasecurity/trivy-operator/pull/232

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

josedonizetti reviewed (commented):
Wow! This is pretty cool. Thanks @erikgb and @padlar 

Left two minor comments. I think not having `version` is not a blocker. Not sure what @chen-keinan thinks.

erikgb commented:
Thanks for the review @josedonizetti, I have addressed both your comments.

chen-keinan commented:
Is pr #232 is duplicate to this one ?

erikgb commented:
> Is pr #232 is duplicate to this one ? is so lets close #232

This PR succeeds #232. I rebased, squashed and filled out the missing pieces after self-review. But a lot of credits to @padlar...
